### PR TITLE
:bug: support to get source from share topic

### PR DESCRIPTION
**Pull request by skeeey**
## Related issue(s)

Fixes #

### CODE REVIEW

File: pkg/cloudevents/generic/options/mqtt/options.go [hunk 1]
```diff
@@ -252,8 +252,18 @@ func validateTopics(topics *types.Topics) error {
 func getSourceFromEventsTopic(topic string) (string, error) {
 	subTopics := strings.Split(topic, "/")
 
+	// get source form share topic, e.g. $share/group/sources/+/consumers/+/agentevents
+	if subTopics[0] == "$share" {
+		if len(subTopics) != 7 {
```
Comment (qiujian16) [inline pkg/cloudevents/generic/options/mqtt/options.go:257]: the number is hard coded? what if the user use a different topic? can we use regex?

Comment (skeeey) [reply pkg/cloudevents/generic/options/mqtt/options.go:257]: The topic length is fixed, but using regex is more grace, will update this

**Review by qiujian16 — COMMENTED** [review]
/approve
/lgtm

**Pull request merged**